File: fixtures/scale/javascript/large/src/ciphers/pulsestream.js
// Keystream cipher (pulsestream) driven by a small LCG.
import bytes from '../util/bytes.js';

function pulsestreamEncrypt(text, key) {
  let x = (key * 7 + 239) % 256;
  const out = bytes.toCodes(text).map((v) => {
    x = (5 * x + 239) % 256;
    return v ^ x;
  });
  return bytes.fromCodes(out);
}

function pulsestreamDecrypt(text, key) {
  return pulsestreamEncrypt(text, key);
}

export default { pulsestreamEncrypt, pulsestreamDecrypt };

File: fixtures/scale/lua/large/ciphers/pulsestream.lua
local bytes = require("util.bytes")

local function pulsestream_encrypt(text, key)
  local x = (key * 7 + 239) % 256
  local codes = bytes.to_codes(text)
  for i, c in ipairs(codes) do
    x = (5 * x + 239) % 256
    codes[i] = c ~ x
  end
  return bytes.from_codes(codes)
end

local function pulsestream_decrypt(text, key)
  return pulsestream_encrypt(text, key)
end

return { pulsestream_encrypt = pulsestream_encrypt, pulsestream_decrypt = pulsestream_decrypt }
